### DISCUSSION TITLE
QuickLook fails to load model displaying "Object requires a newer version of iOS."

### DISCUSSION THREAD

**Discussion by dbbrooks-lgtm** [google/model-viewer#2245] [Q&A]
### Description
QuickLook launches, but fails to load model, and displays the following message: "Object requires a newer version of iOS." 
Devices with repro: iPhone
This must be intermittent because I encountered it a few weeks ago, but it stopped reproducing before I logged a bug. 
Device: iPhone 12 pro
iOS version: 14.4.1 (latest updates installed)

Open Safari and go to https://modelviewer.dev/examples/augmentedreality/#customButton
Click on custom button ("👋 Activate AR") to launch QuickLook

Screenshot:
![Scene-viewer-custom-button-example-issue](https://user-images.githubusercontent.com/77503048/110715963-18ceb700-81bb-11eb-874c-b16484a0b340.jpg)
### Version

- model-viewer: v1.6

### Browser Affected

- [ ] Chrome, version <>
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [x] Safari
### OS

- [ ] Android <>
- [x] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

**Comment by elalish** [comment]
Interesting, but that's not SceneViewer, it's QuickLook (since it's iOS). I'm guessing there must be some kind of Safari problem happening here. Does it repro consistently?

**Comment by dbbrooks-lgtm** [comment]
Oops, I spaced on the name.
It's intermittency is a difficult question to answer. Both times I've encountered it, (a week or two apart) it reprod consistently. But in-between that time, on the same config that I tested the first time I encountered it, it did not repro.

**Comment by russkiimike** [comment]
I found that if the 3d models weren't properly formatted or they were too big, quick look won't work. Always try with the default Astronaut model and if it works then it's an issue with your model (most likely)

**Reply by heikog** [reply]
I have the same provlem here
https://kulturschokolade.org/error/
What do you mean by "properly formatted" ?

**Comment by vortice3D** [comment]
Hi @dbbrooks-lgtm and @heikog:

I'm experiencing this same problem with .reality files opened with Quick Look AR on Safari, under iOS 17. Same contents are shown flawless when visiting the web page on iOS 16 and below.

Any success with the resolution of this malfunction?

Thanks for your time!

**Comment by vortice3D** [comment]
Hi everybody:

Hosting the "CosmonautSuit_en.reality" example from Apple (https://developer.apple.com/augmented-reality/quick-look/) in my server, the same "Object requires a newer version of iOS." is shown on screen.

This behavior leads me to think that the problem can come from an extra security specification for the hosting server of these AR intended files, at least on iOS17 (but not on previous versions).

The case is that I have at the moment a valid SSL/TLS certificate, so I wonder if HSTS would be also necessary.

Thanks for your time.

**Comment by elhamshari** [comment]
I was having this issue , the problem in IOS .
Rebuild your model with texture size 4096 or less .
I hope this help .

**Comment by vortice3D** [comment]
Hi there:

In my case, the problem was server-side, due to a recent change of the MIME type for .reality files, from the previously "application/octet-stream" to the now mandatory with iOS17+ "model/vnd.reality".

Best regards.

**Reply by ZhilinAR** [reply]
I also changed the MIME on my server. Now everything plays correctly on iOS 17, younger iOS versions do not play the .reality file correctly. Can you tell me what to do with this?

**Reply by elalish** [reply]
I'd love to know Apple's recommendation here @grorg.

**Reply by samionsuwito** [reply]
Does anyone know whether you can change the MIME type through the frontend instead? Thanks for your time!

**Reply by rudolphmax** [reply]
@samionsuwito no, generally not. Since MIME-Types are sent by the server, they are configured there as well. If you are using some asset-handling service (like the one AFrame provides), you might be lucky. But typically, servers decide which MIME-Type to use by simply looking at the file's extension. If running an Apache server, extra MIME-Types can be configured in an _.htaccess_ file (`<IfModule mod_expires.c> AddType model/vnd.reality reality </IfModule>`). For Nginx, config-files have to be edited. How exactly depends on the existing config of the server.

For further information, look at the following links
- https://developer.mozilla.org/en-US/docs/Learn/Server-side/Configuring_server_MIME_types#how_to_set_up_your_server_to_send_the_correct_mime_types
- https://www.slingacademy.com/article/nginx-mime-types-the-complete-guide/

**Reply by elalish** [reply]
FYI @AdaRoseCannon

**Reply by AdaRoseCannon** [reply]
Thanks for bringing this to my attention @elalish
